packages/skygear-web/src/container.ts

```diff
@@ -207,60 +243,49 @@ export class WebAuthContainer<T extends WebAPIClient> extends AuthContainer<T> {
     window.location.href = url;
   }
 
-  private async _getRedirectResult(action: "login" | "link"): Promise<any> {
-    if (this.oauthResultObserver == null) {
-      this.oauthResultObserver = new WindowMessageObserver(
-        this.parent.apiClient.endpoint
-      );
+  private async _getRedirectResult(
+    action: "login" | "link"
+  ): Promise<string | null> {
+    const lastAction = await this.parent.storage.getOAuthRedirectAction(
+      this.parent.name
+    );
+    if (lastAction !== action) {
+      return null;
     }
 
-    let iframe: HTMLIFrameElement | undefined;
-    try {
-      const lastAction = await this.parent.storage.getOAuthRedirectAction(
-        this.parent.name
-      );
-      if (lastAction !== action) {
-        return undefined;
-      }
-
-      await this.parent.storage.delOAuthRedirectAction(this.parent.name);
-      const messagePromise = this.oauthResultObserver.subscribe();
+    await this.parent.storage.delOAuthRedirectAction(this.parent.name);
 
-      iframe = document.createElement("iframe");
-      iframe.style.display = "none";
-      iframe.src = this.parent.apiClient.endpoint + "/_auth/sso/iframe_handler";
-      document.body.appendChild(iframe);
-
-      const message = await messagePromise;
-      return decodeMessage(message);
-    } finally {
-      if (iframe) {
-        this.oauthResultObserver.unsubscribe();
-        document.body.removeChild(iframe);
-      }
+    const searchParams = new URLSearchParams(window.location.search.slice(1));
+    const result = searchParams.get("x-skygear-result");
```

Comment:
What do you think about erasing the parameter after reading it using `replaceState`? So that it would not leave potentially sensitive information in browser history.

Reply:
I am afraid that it may break the application. The application may be using a library to manipulate history and does not expect the actual history to be modified directly.

Reply:
I agree with your concern. I suppose we'll mention in the documentation that, the developer should erase the query parameters for better security.